### DISCUSSION TITLE
[Prettier Plugin] order classes for lg screen get putted first

### DISCUSSION THREAD

**Discussion by Faycop** [tailwindlabs/tailwindcss#14372] [Help] [answered]
I have this tailwind.config.js file:

`/** @type {import('tailwindcss').Config} */

const defaultTheme = require('tailwindcss/defaultTheme')
const plugin = require('tailwindcss/plugin');

// This file controls how Tailwind processes your CSS. For details, see
// https://tailwindcss.com/docs/configuration

module.exports =
{
  //
  // WARNING: By default, CodeKit automatically populates the `content` array with all entries from [Project Settings > PurgeCSS]
  // in CodeKit's UI. If you add ANY entries to the `content` array here, CodeKit will not auto-populate the array; it becomes your
  // responsibility to include every type of file in your project that uses CSS rules. It is preferable to edit the PurgeCSS content 
  // list in CodeKit's UI.
  // 
  // WARNING: DO NOT delete `content` or comment it out. If you do, CodeKit will treat this as a Tailwind 2.x project instead of 3.x. 
  //
  content: [],

  //
  // All other TailwindCSS options are 100% under your control. Edit this config file as shown in the Tailwind Docs
  // to enable the settings or customizations you need.
  // 
  theme: {
    extend: {
      padding: {
        '1/2': '50%',
        full: '100%',
      },
      fontFamily: {
        heading: ['Poppins', ...defaultTheme.fontFamily.sans],
        sans: ['Lato', ...defaultTheme.fontFamily.sans],
      },
      typography: ({ theme }) => ({
        DEFAULT: {
          css: {
            color: theme('colors.content'),
            a: {
              color: theme('colors.primary[DEFAULT]'),
              '&:hover': {
                color: theme('colors.secondary'),
              },
            },
            pre: {
              'line-height': '1'
            }
          },
        },
      }),
      fontSize: {
        'caption': ['0.75rem', { // 12px
          lineHeight: '1rem',
          fontWeight: '400',
        }],
        'sm': ['0.875rem', { // 14px
          lineHeight: '1.25rem',
          fontWeight: '400',
        }],
        'md': ['1rem', { // 16px
          lineHeight: '1.5rem',
          fontWeight: '400',
        }],
        'lg': ['1.125rem', { // 18px
          lineHeight: '2.125rem',
          fontWeight: '400',
        }],
        'base': ['1rem', { // 16px - h6/p
          lineHeight: '1.5rem',
          fontWeight: '400',
        }],
        'xl': ['1.25rem', { // 20px - h5
          lineHeight: '2rem',
          fontWeight: '500',
        }],
        '2xl': ['1.5rem', { // 24px - h4
          lineHeight: '2.5rem',
          letterSpacing: '-0.01em',
          fontWeight: '500',
        }],
        '3xl': ['2rem', { // 32px - h3
          lineHeight: '2.95rem',
          letterSpacing: '-0.01em',
          fontWeight: '600',
        }],
        '4xl': ['2.5rem', { // 40px - h2
          lineHeight: '3.5rem',
          letterSpacing: '-0.01em',
          fontWeight: '600',
        }],
        '5xl': ['3rem', { // 48px - h1
          lineHeight: '4.25rem',
          letterSpacing: '0.005em',
          fontWeight: '600',
        }],
        'display': ['3.5rem', { // 56px
          lineHeight: '5rem',
          letterSpacing: '0.005em',
          fontWeight: '600',
        }],
      },
      colors: {
        'secondary': {
          DEFAULT: '#004D40',
          'letter': '#525252'
        },
        'primary': {
          DEFAULT: '#4CAF50',
          'letter': '#161616',
          100: '#223F21',
          200: '#2C552B',
          300: '#346B34',
          400: '#3B7D3D',
          500: '#4CAF50',
          600: '#62B762',
          700: '#90CB8C',
          800: '#B9DFB5',
          900: '#DAEED7',
          950: '#ECF6EB'
        },
        'accent': {
          DEFAULT: '#CDDC39',
          100: '#383A19',
          200: '#4C4F1E',
          300: '#5E6323',
          400: '#6E7426',
          500: '#88912C',
          600: '#A1AC31',
          700: '#B8C635',
          800: '#D0DE48',
          900: '#EBEDA0',
          950: '#F6F6D0'
        },
        'scopri': {
          DEFAULT: '#FF5722',
          'title': '#FF6E40',
          'cta': '#CDDC39',
          100: '#582D09',
          200: '#7A3B08',
          300: '#9A4705',
          400: '#B85102',
          500: '#E96100',
          600: '#F78748',
          700: '#FBAB7F',
          800: '#FDCAAF',
          900: '#FFE3D5',
          950: '#FFF0EA'
        },
        'impara': {
          DEFAULT: '#9C27B0',
          'title': '#EA80FC',
          'cta': '#CDDC39',
          100: '#591F62',
          200: '#7B238A',
          300: '#9B27AF',
          400: '#A844B9',
          500: '#BC6DC7',
          600: '#CD8FD4',
          700: '#DCAEE0',
          800: '#E9CBEC',
          900: '#F4E3F5',
          950: '#FAF1FA'
        },
        'partecipa': {
          DEFAULT: '#512DA8',
          'title': '#B388FF',
          'cta': '#CDDC39',
          100: '#133672',
          200: '#09489F',
          300: '#005ACA',
          400: '#3169DF',
          500: '#7184E6',
          600: '#999FEC',
          700: '#B8B8F1',
          800: '#D3D1F6',
          900: '#E8E6FA',
          950: '#F4F2FD'
        },
        'neutral-100': '#33393C', // Overrided this way because of DaisyUI
        'neutral-200': '#454E52', // Overrided this way because of DaisyUI
        'neutral-300': '#566167', // Overrided this way because of DaisyUI
        'neutral-400': '#647278', // Overrided this way because of DaisyUI
        'neutral-500': '#7D8E96', // Overrided this way because of DaisyUI
        'neutral-600': '#95A8B1', // Overrided this way because of DaisyUI
        'neutral-700': '#B1BFC6', // Overrided this way because of DaisyUI
        'neutral-800': '#CDD6DB', // Overrided this way because of DaisyUI
        'neutral-900': '#E4E9EB', // Overrided this way because of DaisyUI
        'neutral-950': '#F1F4F5', // Overrided this way because of DaisyUI
      }
    }
  },

  variants: {},

  //
  // If you want to run any Tailwind plugins (such as 'tailwindcss-typography'), simply install those into the Project via the
  // Packages area in CodeKit, then pass their names (and, optionally, any configuration values) here. 
  // Full file paths are not necessary; CodeKit will find them.
  //
  plugins: [
    require('@tailwindcss/aspect-ratio'),
    require('@tailwindcss/container-queries'),
    require('@tailwindcss/forms'),
    require('@tailwindcss/typography'),
    require('tailwindcss-animated'),
    require('tailwind-scrollbar-hide'),
    require('tailwindcss-opentype'),
    require('tailwindcss-3d'),
    require('daisyui'),

    plugin(function ({ addVariant }) {
      addVariant('progress-unfilled', ['&::-webkit-progress-bar', '&']);
      addVariant('progress-filled', ['&::-webkit-progress-value', '&::-moz-progress-bar', '&']);
    })
  ],

  daisyui: {
    themes: [
      {
        bsg: {
          "primary": "#4CAF50",
          "primary-content": "#161616",
          "secondary": "#004D40",
          "secondary-content": "#F1F4F5",
          "accent": "#CDDC39",
          "accent-content": "#161616",
          "neutral": "#CDD6DB",
          "neutral-content": "#161616",
          "base-100": "#F1F4F5",
          "base-200": "#E4E9EB",
          "base-300": "#CDD6DB",
          "base-content": "#33393C",

          "info": "#1192E8",
          "info-content": "#F1F4F5",
          "success": "#198038",
          "success-content": "#F1F4F5",
          "warning": "#F1C21B",
          "warning-content": "#161616",
          "error": "#da1e28",
          "error-content": "#F1F4F5",

          "--rounded-box": "1rem", // border radius rounded-box utility class, used in card and other large boxes
          "--rounded-btn": "0.5rem", // border radius rounded-btn utility class, used in buttons and similar element
          "--rounded-badge": "1.9rem", // border radius rounded-badge utility class, used in badges and similar
          "--animation-btn": "0.25s", // duration of animation when you click on button
          "--animation-input": "0.2s", // duration of animation for inputs like checkbox, toggle, radio, etc
          "--btn-focus-scale": "0.95", // scale transform of button when you focus on it
          "--border-btn": "1px", // border width of buttons
          "--tab-border": "1px", // border width of tabs
          "--tab-radius": "0.5rem", // border radius of tabs

          ".btn-twitter": {
            "background-color": "#1EA1F1",
            "border-color": "#1EA1F1",
          },
          ".btn-twitter:hover": {
            "background-color": "#1C96E1",
            "border-color": "#1C96E1",
          },
        },
      },
    ],
  },
}`

my prettierrc file is this:

`{
  "plugins": ["prettier-plugin-tailwindcss"]
}`

The problem is that when I use the `lg` screen class prettier puts the class first and then I receive this error:

`CssSyntaxError: /Users/flavio/Development/BSG/render/source/css/main.css:66:5: The `lg:p-30` class does not exist. If `lg:p-30` is a custom class, make sure it is defined within a `@layer` directive.

  64 |   }
  65 |   .fundamentals-body {
> 66 |     @apply lg:p-30 container mx-auto bg-white p-10 text-primary-letter md:p-20 xl:p-40;
     |     ^
  67 |   }
  68 | }`

As you can see the lg screen is never overrided in the tailwind config file. Why prettier doesn't recognize only the lg screen?

Edit: Also, using the tailwind intellisense, if I hover on the (example) md screen on my markup I see what I get:

<img width="296" alt="Screenshot 2024-09-09 alle 16 57 14" src="https://github.com/user-attachments/assets/d2c5859b-a548-4ff7-a08d-f1a1d8bfaa5c">

For the lg screen instead I get nothing:

<img width="160" alt="Screenshot 2024-09-09 alle 16 57 22" src="https://github.com/user-attachments/assets/62d93626-7dd2-4c34-ae45-547a0e77e165">

**Comment by adamwathan** [comment]
Hey! Like the error mentions, there is no `p-30` class in Tailwind by default.

You can add it if you need it by customizing your configuration:

https://tailwindcss.com/docs/customizing-spacing#extending-the-default-spacing-scale

**Reply by Faycop** [reply, marked as answer]
Ouch, you are ABSOLUTELY RIGHT. As intellisense suggests, there is no class p-30 available:

<img width="466" alt="Screenshot 2024-09-09 alle 19 07 51" src="https://github.com/user-attachments/assets/f4d7192a-6caf-4ca1-9771-0b8802583a85">

Using class `p-32` goes smooth as butter. Thanks and sorry for the useless issue!
